include/unix/osd.h

```diff
@@ -304,7 +306,46 @@ ofi_cpuid(unsigned func, unsigned subfunc, unsigned cpuinfo[4])
 	asm volatile("clflush %0" : "+m" (*(volatile char *) addr))
 #define ofi_sfence() asm volatile("sfence" ::: "memory")
 
-#else /* defined(__x86_64__) || defined(__amd64__) */
+#elif ( defined(__riscv) && defined(__riscv_xlen) && (__riscv_xlen == 64) && (defined(__linux__) || defined(__APPLE__)) )
+
+#include <stdio.h>
+#include <regex.h>
+
+static inline void
+ofi_cpuid(unsigned func, unsigned subfunc, unsigned cpuinfo[4])
+{
+     FILE *f = fopen("/proc/cpuinfo", "r");
+
+     regex_t r;
+     regmatch_t  match[1];
+
+     regcomp(&r, "hart", 0);
+     unsigned int hart = 0;
+     size_t lsz = 1024;
+     char *line = malloc(1024);
+     size_t rc = getline(&line, &lsz, f);
+     while(rc != -1 && 0 < rc) {
+         if(!regexec(&r, line, sizeof(match) / sizeof(match[0]), match, 0)) {
+            hart+=1;
+         }
+         
+         rc = getline(&line, &lsz, f);
+     }
+
+     regfree(&r);
+
+     cpuinfo[0] = hart & 0x000000FF;
+     cpuinfo[1] = hart & 0x0000FF00;
+     cpuinfo[2] = hart & 0x00FF0000;
+     cpuinfo[3] = hart & 0xFF000000;
+}
```

Comment:
It's nice to have this function, but is it actually needed, and does it work?  ofi_cpuid() is only called from ofi_cpu_supports(), which is used for persistent memory.  Do the risc-v cpuinfo bits align with x86_64 on those bits?  I'm going to guess not, because the calls below are no-ops (ofi_clwb, ofi_clflushopt, etc)

Reply:
It was implemented under the impression it would be necessary. I'll remove it.